[hardknott] audit 2-5: recipes-ni changes

amstewart opened the pull request:
This PR adds some missing runtime dependencies to recipes in the `recipes-ni/` directory. It also moves the `fw-printenv` reciep out of the `ni-utils/` subdirectory, into its own.

# Testing
* `fw-printenv` still builds (which verifies the above changes, since they don't substantially impact recipe logic.)
* The other changes in this PR are adding common packages to `RDEPENDS`, so I expect that they will all reliably install on a running target.

amstewart committed:
ni-smbios-helper: add missing RDEPENDs

The scripts installed by ni-smbios-helper call dmidecode and efivar as
external tools.

Add RDEPENDs on both of these utilities to ensure that they are
available when the scripts are run.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemformat: add missing RDEPENDs

The nisystemformat script calls some external dependencies which deserve
RDEPENDs:
* lsblk from util-linux-lsblk
* uname from coreutils
* ninetcfgutil from ni-netcfgutil
* mkfs.ext4 from e2fsprogs-mke2fs

Add RDEPENDs on those packages.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
ni-systemimage: add missing RDEPENDS

The nisystemimage script depends on the find utility from findutils and
the functions.common source file from ni-utils.

Add RDEPENDS on those utils.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
fw-printenv: move recipe to its own dir

The fw-printenv recipe is substantial enough that it warrants its own
directory.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart committed:
niwatchdogpet: add missing RDEPENDS

The niwatchdogpet initscript calls fw_printenv.

Add an RDEPENDS on fw_printenv to the recipe.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

shruthi-ravi commented on recipes-ni/ni-systemformat/ni-systemformat.bb:
```diff
@@ -42,4 +42,14 @@ inherit update-rc.d
+	ni-netcfgutil \
+	util-linux-lsblk \
+"
+RDEPENDS:${PN}:append:x64 = " e2fsprogs-mke2fs"
```
Why a separate x64 list here?

amstewart replied:
The `nisystemformat` script [uses the `mkfs.ext4` binary](https://github.com/ni/meta-nilrt/blob/nilrt/master/hardknott/recipes-ni/ni-systemformat/files/nisystemformat#L172), but only when the system architecture is x64 (for the obvious reasons). So the runtime dependency technically only exists on x64.

shruthi-ravi replied:
I thought we were removing x64-specific code with the assumption that we support this image only on x64. Is that not the case here?

amstewart replied:
I originally didn't expect that `e2fsprogs-mke2fs` would exist in the ARM feeds, but I suppose we don't build those in hardknott anyway, so it doesn't matter. Will fix.

amstewart replied:
Fixed in https://github.com/ni/meta-nilrt/pull/483/commits/57a560de271c9ac09ee990368e98a9080279226c.

amstewart committed:
ni-systemformat: collapse recipe RDEPENDS

There is no reason to break out some recipe RDEPENDS for x64, since it
is the only supported architecture in the hardknott mainline.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
